app/src/main/java/com/poupa/vinylmusicplayer/preferences/SongConfirmationPreference.java

```diff
@@ -0,0 +1,40 @@
+package com.poupa.vinylmusicplayer.preferences;
+
+
+import java.util.Arrays;
+import java.util.List;
+
+import android.content.Context;
+import android.util.AttributeSet;
+
+import com.kabouzeid.appthemehelper.common.prefs.supportv7.ATEDialogPreference;
+import com.poupa.vinylmusicplayer.R;
+import com.poupa.vinylmusicplayer.dialogs.BottomSheetDialog.BottomSheetDialogWithButtons;
```

Comment:
Hmm, I'd avoid having the `.preferences.*` code depending on the `.dialogs.*` (the reverse would be okay)

Probably we can migrate the `.ButtonInfo` class here, as a new base class `SongConfoPref.Choice`.

Reply:
But this means that if we want to convert all dialog to bottomsheet one later, they will all depend on SongConfirmationPref, seems really strange to me (more so because the implementation of buttoninfo is only constrained by bottomsheetwithbutton, not any preference class).
  
In fact it just append that it seems visually cleaner for this preference to use the same dialog layout than the confirmation one for user clarity. This preference could have also use a more basic radio button list view like other pref (I wanted to get same icon in the confirmation and in the pref).

Reply:
In fact this new class (SongConfirmationPreference) is a UI helper class to manipulate the prefs. I did not understand that way - that's why I had made the comment.

The `.preferences` package already contain a bunch of dialog implementation for different settings --> OK to proceed